### DISCUSSION TITLE
What ZigBee Adapter to buy that has NO quirks?

### DISCUSSION THREAD

**Discussion by maichai** [Koenkk/zigbee2mqtt#6583] [Devices]
I currently use a Texas Instruments LAUNCHXL-CC26X2R1. The problem with this device is that from time to time I need to power-cycle it manually to get it back to work properly. I learned that this is a known issue. I wish, since this is known, that it would also have been documented in the list of the supported controllers, but that is not the case up until today. This should be done as a courtesy to future buyers.

Now I am fed up with this controller problem and I want to buy a new controller. However, since the known problems do not seem to be fully documented in the list of the supported controllers, I would like to make sure up front to not buy a controller that also has known problems.

The controller should be 
- flashable without additional hardware, 
- work as a coordinator and as a router, 
- run completely unattended with no need for manual attention and 
- should start and work properly after the host reboots
- support a lot more than 20 devices
- has enough power to reach into further rooms in a large appartment

Think of it as a device that will be far away and very hard to access manually after it has been installed.

Currently on my candidate list are:

- Texas Instruments LAUNCHXL-CC1352P-2
- Electrolama zig-a-zig-ah! (zzh!) 
- Slaesh's CC2652RB stick 

Which of these devices meet my criteria?

**Comment by Koenkk** [comment]
I recommend either the ZZH or CC2652RB (there are no differences for the user, see point 3 of https://www.zigbee2mqtt.io/information/supported_adapters.html)

**Reply by InvertedMinor** [reply]
do not buy from Slaeh, he accpets your money but no communication I am still waiting for my usb stick

**Comment by maichai** [comment]
I saw that, but that does not address my point. 

I have a bunch of criteria given and I would love someone (not necessarily you) to actually contribute their experiences regarding the candidate controllers - or one of them - with regards to my criteria.

**Comment by diplix** [comment]
i have a CC26X2R1 (and a network of around 70 devices), too and experienced the same quirk where i had to powercycle the device almost every week. as a workaround i hooked it up to a dedicated raspberry pi so that i could automatically trigger a restart of the pi, which also powercycled the CC26X2R1.

since release [1.17.1](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.17.1) with the latest coordinator [firmware update](https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin) these problems have gone. the firmware update included a fix for a memory leak (aka the known issue) from ti ([also see this](https://www.reddit.com/r/homeassistant/comments/l9uhvm/psa_zigbee_network_lockup_bug_affecting/)). my pi and the connected CC26X2R1 now have been up without needing to powercycle since i upgraded the firmware (~6 weeks).

so i’d say, with the recent coordinator firmware update the CC26X2R1 for me is quirk-free.

**Reply by maichai** [reply]
I will give this a shot tomorrow

**Reply by WarmCatUK** [reply]
Thanks for this info, I have a CC1352P-2 and did experience an incident when nothing was responding, not sure if it was related as I'd been adding devices and changing stuff around the same time.
Anyway I guess I should upgrade firmware, is there any precautions I should take first?

**Comment by ccorderod** [comment]
@maichai I can't say re zzh or CC2652RB stick. I use 2 Texas Instruments LAUNCHXL-CC1352P-2, supporting a network well over 100 devices each, spread on 3 floors some of the devices far away (50m) from the coordinator. Each coordinator is connected to a Pi4 on sd card using a usb extension cable (1.5m). I have been using those after migrating from ccxx31 sticks running the ZB 1.x for more than 2 years now. NO COMPLAIN, with the exception of the hue motion sensors which still show red lights from time to time.
Devices connected are a mix of routers (hue lights, ikea lights, xiaomi gas sensors, meazon, xiaomi and innr meters a few ccxx31 routers) and end devices (aqara door/window, vibration and waterleak sensors, aqara cubes, nyce door and motion , hue motion in and outdoor, ikea dimmers and remote, hue remotes). All are supported by Z2M.
Those coordinator are flashable through usb using ti software, work as coordinator and router (no experience with the routing software), run unattended (since 2 years, with the exception of firmware update from time to time), no issues booting/rebooting and you may setup them on 20dB mode).
Those are my 2 cts.

**Reply by maichai** [reply]
If my CC26X2R1 does not lose the problem after the firmware update i will switch to this platform. Thanks for your long response.

**Reply by maichai** [reply]
Shame on me, I found that I actually own a LAUNCHXL-­CC1352P-­2: I tried the latest firmware for the launchpad, but z2m couldn't connect to it, so I rolled back to the previous firmware from 2019.

Is there anythign special that needs to be done in the z2m config file when migrating the latest firmware?

what is the latest working firmware version you are running on your launchpads?

**Reply by Shred99** [reply]
I migrated everything (22 sensors) to a LAUNCHXL-CC1352P-2 about six months ago - it might be longer.  It's pretty much rock solid.  I've never had to restart the Raspberry Pi that runs it, or to disconnect/re-connect the board.  I upgraded to the current firmware on the weekend and didn't have to change anything or do anything special.

Check that your serial port hasn't changed when you upgraded the firmware.  The LAUNCHXL boards are presented to the host computer as two serial ports - it may be possible that the OS has swapped the names of the serial ports?

**Comment by maichai** [comment]
The device name did change, but I had seen that and using the new name it still didn't work. Reason was that I used the router firmware, not the coordinator firmware. Now using the coordinator firmware it is working fine.

**Comment by m0wlheld** [comment]
> The controller should be
> 
> * flashable without additional hardware,
> * work as a coordinator and as a router,
> * run completely unattended with no need for manual attention and
> * should start and work properly after the host reboots
> * support a lot more than 20 devices

Slaesh's CC2652RB stick never failed on me, with the exception that when it is offline for a longer period of time (15min+), Zigbee2MQTT won‘t properly start, claiming PanID is already taken. 

Then I have to power down all routing devices (bulbs, sockets, etc) and restart. After that, I can re-power all devices again, no re-pair needed. 

> * has enough power to reach into further rooms in a large appartment

You should reconsider this criterion, that‘s not how Zigbee works.  You should make your nextwork more dense with devices on mains or dedicated routers.

**Comment by Jim-Noble** [comment]
I have a zzh stick. My experience has been virtually flawless, and would meet all your criteria. 
I've had one short period of instability of the network after I made a whole bunch of simultaneous changes - new z2m version, new firmware, and (probably most significantly) moved the stick location from the end of a 2m USB extension to plugged into the raspberry Pi directly. 
This resulted in the network falling apart for a few days. It's back to rock solid stable now after a couple of days of having to reset/re-add devices.  
I have ~50 devices, including 4 ikea repeater routers, 2 ikea sockets, plus one ikea lamp, and one hue lamp which aren't powered all the time.
Even though I have the extra routers, ~9 of the non-router devices connect directly to the coordinator, some of which are at the far reaches of the [3 storey, ~200m^2] house (plus one outside). 
I also run a separate zigbee network using my original CC2531 coordinator with smartthings sockets, as they don't play nice with the xiaomi devices (nothing to do with zzh - that also affected the original coordinator). HTH...
